app/controllers/merchants_items_controller.rb

```diff
@@ -0,0 +1,9 @@
+class MerchantsItemsController < ApplicationController
+   def index
+    @merchant = Merchant.find(params[:merchant_id])
+  end
+
```

Comment:
think this is a more restful solution than having the items controller taking every merchants items action